components/project/branch.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import Div from 'components/core/div'
+import styled from 'styled-components'
+
+const BranchContainer = styled(Div)`
+  justify-content: space-between;
+`
+
+const ValueContainer = styled(Div)`
+  color: ${props => (props.header ? props.theme.colors.green : props.theme.colors.white)};
+`
+
+const BranchNameContainer = styled(ValueContainer)`
+  width: 300px;
+  display: block;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+`
+
+const TimeContainer = styled(ValueContainer)`
+  margin-left: ${props => props.theme.spaces.extraSmall}px;
+  width: 270px;
+`
+
+const StatusContainer = styled(ValueContainer)`
+  width: 100px;
+  ${props => (props.failed ? `color: ${props.theme.colors.red};` : '')};
+`
+
+export default props => {
+  const latestBuild = props.branch.latestBuild
+
+  return (
+    <BranchContainer>
+      <BranchNameContainer header={props.header}>{props.branch.name}</BranchNameContainer>
```

Comment:
following on from my other common. this `if` `else` has now permeated several other components.

and what you're kind of saying is: 'Am I in this mode' ok I'll do thing A) 'or am in that mode' I'll do thing B). Once you refactor this code you'll see what it looks like and it will look much better